Index extraction/optimization scripts fail during upgrade to OC 18.x on Debian

During upgrading from Opencast 17.10 to 18.4 on a Debian 13 system I run into issues while running the index migration scripts:

Both [extract_es_data.sh](https://github.com/opencast/opencast/blob/r/18.x/docs/upgrade/17_to_18/extract_es_data.sh) and [optimize_search.sh](https://github.com/opencast/opencast/blob/r/18.x/docs/upgrade/17_to_18/optimize_search.sh) refuse to run with
```
Error: jq is not installed. Please install it and try again.
```
while `jq` has been installed just minutes before.

Turns out, despite using `#!/bin/sh` the scripts are not POSIX compliant but require Bash.
On Debian, `/bin/sh` points do Dash, which fails running the scripts.

Shellcheck (e.g. on https://www.shellcheck.net/) yields different issues:

`extract_es_data.sh`:
```
[Line 4:](javascript:setPosition(4, 8))
set -o pipefail
       ^-- [SC3040](https://www.shellcheck.net/wiki/SC3040) (warning): In POSIX sh, set option pipefail is undefined.

[Line 20:](javascript:setPosition(20, 20))
if ! command -v jq &> /dev/null; then
                   ^-- [SC3020](https://www.shellcheck.net/wiki/SC3020) (warning): In POSIX sh, &> is undefined.
```

`optimize_search.sh`:
```
[Line 4:](javascript:setPosition(4, 8))
set -o pipefail
       ^-- [SC3040](https://www.shellcheck.net/wiki/SC3040) (warning): In POSIX sh, set option pipefail is undefined.
 
[Line 13:](javascript:setPosition(13, 20))
if ! command -v jq &> /dev/null; then
                   ^-- [SC3020](https://www.shellcheck.net/wiki/SC3020) (warning): In POSIX sh, &> is undefined.
 
[Line 68:](javascript:setPosition(68, 8))
    if [[ "$completed" == "true" ]]; then
       ^-- [SC3010](https://www.shellcheck.net/wiki/SC3010) (warning): In POSIX sh, [[ ]] is undefined.
                       ^-- [SC3014](https://www.shellcheck.net/wiki/SC3014) (warning): In POSIX sh, == in place of = is undefined.
 
[Line 102:](javascript:setPosition(102, 8))
    if [[ "$completed" == "true" ]]; then
       ^-- [SC3010](https://www.shellcheck.net/wiki/SC3010) (warning): In POSIX sh, [[ ]] is undefined.
                       ^-- [SC3014](https://www.shellcheck.net/wiki/SC3014) (warning): In POSIX sh, == in place of = is undefined.
 
[Line 128:](javascript:setPosition(128, 33))
  if (ctx._source.containsKey('"$FIELD_NAME"_fuzzy')) {
                                ^-- [SC2027](https://www.shellcheck.net/wiki/SC2027) (warning): The surrounding quotes actually unquote this. Remove or escape them.
                                           ^-- [SC1078](https://www.shellcheck.net/wiki/SC1078) (warning): Did you forget to close this double quoted string?
 
[Line 129:](javascript:setPosition(129, 18))
    ctx._source['"$FIELD_NAME"_fuzzy'] = [];
                 ^-- [SC1079](https://www.shellcheck.net/wiki/SC1079) (info): This is actually an end quote, but due to next char it looks suspect.
                  ^-- [SC2027](https://www.shellcheck.net/wiki/SC2027) (warning): The surrounding quotes actually unquote this. Remove or escape them.
```

I guess the easiest would be to simply use `#!/bin/bash` instead. While fixing the issue and avoiding some confusion for other users it should work on all supported Linux distros (RHEL/Rocky/Ubuntu/Debian)?

--- a/docs/upgrade/17_to_18/extract_es_data.sh
+++ b/docs/upgrade/17_to_18/extract_es_data.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 set -e
 set -o pipefail

--- a/docs/upgrade/17_to_18/optimize_search.sh
+++ b/docs/upgrade/17_to_18/optimize_search.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 set -ex
 set -o pipefail

